lib/components/message_input.dart

```diff
@@ -206,30 +90,75 @@ class _MessageInputWidgetState extends State<MessageInputWidget> {
       commandsModel.addCommand(Command(value, DateTime.now()));
     }
     ActionsAdapter.instance.send(widget.channel, value);
-    _textEditingController.clear();
+    setState(() => _textEditingController.clear());
   }
 
   Widget _buildEmotePicker(BuildContext context) {
     return EmotePickerWidget(
         channelId: widget.channel.channelId,
-        onDismiss: () => setState(() => _isEmotePickerVisible = false),
-        onDelete: () {
-          var initialText = _textEditingController.text;
-          if (initialText.isNotEmpty) {
-            _textEditingController.text =
-                initialText.substring(0, initialText.length - 1);
-          }
-        },
         onEmoteSelected: (emote) {
-          _textEditingController.text =
-              "${_textEditingController.text} ${emote.code}";
+          setState(() {
```

Comment:
Won't this and the listener cause rerenders when using the emote picker?

Reply:
Yes but we have to trigger a rerender to collapse the command panel